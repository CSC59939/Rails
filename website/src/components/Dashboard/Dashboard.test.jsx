import { Dashboard } from '..';
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Button, Card, Icon, Layout } from 'antd';
configure({ adapter: new Adapter() });
const { Content } = Layout;

describe("SampleComponent", () => {
  it('Expect SampleComponent to return a div with its name', () => { // eslint-disable-line no-undef
    const component = renderer.create(
      <Dashboard> <h1> Hello Word </h1></Dashboard>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot(); // eslint-disable-line no-undef
  });
  it('should contains a h1 tag with Hello World as children of Dashboard', () => {
  expect(shallow(<Dashboard> <h1>Hello World</h1></Dashboard>).contains(<h1>Hello World</h1>)).toBe(true)
});
  it('should have a class name called Container', () => {
   expect(shallow(<Dashboard> <h1>Hello World</h1></Dashboard>).exists('.Container')).toBe(true)
 });
})
