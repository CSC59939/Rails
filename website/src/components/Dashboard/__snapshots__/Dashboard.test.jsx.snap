// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SampleComponent Expect SampleComponent to return a div with its name 1`] = `
<div
  className="Container ant-layout"
  style={
    Object {
      "height": "100%",
    }
  }
>
  <div
    className="Header ant-layout-header"
  >
    <div
      className="ProfileInfo"
    >
      <i
        className="ProfileImage material-icons"
        style={null}
      >
        person
      </i>
      <div
        className="rightSide"
      >
        <p
          className="name"
        >
          User Name
        </p>
        <p
          className="email"
        >
          jdoe@gmail.com
        </p>
      </div>
    </div>
    <div
      className="HeaderIcons"
    >
      <button
        className="IconButton"
        onClick={[Function]}
        type="button"
      >
        <i
          className="material-icons"
          style={null}
        >
          add_circle
        </i>
      </button>
      <button
        className="IconButton"
        onClick={[Function]}
        type="button"
      >
        <i
          className="material-icons"
          style={null}
        >
          add_alert
        </i>
      </button>
      <button
        className="IconButton"
        onClick={[Function]}
        type="button"
      >
        <i
          className="material-icons"
          style={null}
        >
          notifications_active
        </i>
      </button>
      <button
        className="IconButton"
        onClick={[Function]}
        type="button"
      >
        <i
          className="material-icons"
          style={null}
        >
          settings
        </i>
      </button>
    </div>
  </div>
  <div
    className="ant-layout"
    style={
      Object {
        "height": "100%",
      }
    }
  >
    <div
      className="ant-layout-content"
      style={
        Object {
          "height": "100%",
        }
      }
    >
       
      <h1>
         Hello Word 
      </h1>
    </div>
  </div>
</div>
`;
